jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/utility/BluetoothDateTime.kt

```diff
@@ -0,0 +1,72 @@
+package org.ehealthinnovation.jdrfandroidbleparser.bgm.utility
+
+import java.time.Year
+import java.util.*
+
+data class BluetoothDateTime(val _year: Int = 0,
+                             val _month: Int = 0,
+                             val _day: Int = 0,
+                             val _hour: Int = 0,
+                             val _min: Int = 0,
+                             val _sec: Int = 0,
+                             val _offset: Int = 0) {
+    var year: Int
+    var month: Int
+    var day: Int
+    var hour: Int
+    var min: Int
+    var sec: Int
+    var offset: Int
+
+    lateinit var date: Date
+
+    init {
+        year = _year
+        month = _month
+        day = _day
+        hour = _hour
+        min = _min
+        sec = _sec
+        offset = _offset
+    }
```

Comment:
You can do this in the actual constructor statement, and you don't need to do it in the init block

Reply:
move the initialization to the property declarations